File: GameLoseLayer.js

var GameLoseLayer = function () {
    this.delegate = null;
};

GameLoseLayer.prototype.onDidLoadFromCCB = function () {
    var winSize = cc.Director.getInstance().getWinSize();
    this.rootNode.setPosition(cc.p(winSize.width / 2 - this.rootNode.getContentSize().width / 2, winSize.height / 2 - this.rootNode.getContentSize().height / 2));
    cc.Director.getInstance().getTouchDispatcher().addTargetedDelegate(this, cc.MENU_HANDLER_PRIORITY - 1, true);
    this.updateCCBElements();
};

GameLoseLayer.prototype.updateCCBElements = function () {    
    this.titleLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELOSE));
    this.infoLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELOSE_INFO));
    this.restartButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_RESTART_GAME));
    this.mainMenuButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_MAINMENU));
    this.helpButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAME_TUTORIAL));
};

GameLoseLayer.prototype.onHelpPressed = function () {
    if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    this.rootNode.removeFromParent(true);
    var scene = cc.BuilderReader.loadAsScene("HelpScene.ccbi");
    cc.Director.getInstance().pushScene(cc.TransitionFade.create(1.0, scene));
};

GameLoseLayer.prototype.onMainMenuPressed = function () {
	this.rootNode.removeFromParent(true);
    if (this.delegate) this.delegate.toMainMenu();
};

GameLoseLayer.prototype.onRestartGamePressed = function () {
    this.rootNode.removeFromParent(true);
    if (this.delegate) this.delegate.restartGame();
};

File: ClearFiveStages-iOS/ClearFiveStages/Scene/GameLoseLayer.m
#import "GameLoseLayer.h"
#import "GameDataManager.h"
#import "SimpleAudioEngine.h"
#import "MultiLanguageUtil.h"
#import "CCBReader.h"

@implementation GameLoseLayer
@synthesize delegate;

-(void) didLoadFromCCB {
	CGSize winSize = [CCDirector sharedDirector].winSize;
	self.position = ccp(winSize.width / 2 - self.contentSize.width / 2, winSize.height / 2 - self.contentSize.height / 2);
	[self updateCCBElements];
}

-(void) registerWithTouchDispatcher {
	[[CCDirector sharedDirector].touchDispatcher addTargetedDelegate:self priority:kCCMenuHandlerPriority swallowsTouches:YES];
}

-(BOOL) ccTouchBegan:(UITouch *)touch withEvent:(UIEvent *)event {
    return YES;
}

-(void) dealloc {
	if (self.delegate) [self.delegate release];
    [super dealloc];
}

// multi-language handle
-(void) updateCCBElements {
	[titleLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELOSE]];
	[infoLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELOSE_INFO]];
	[restartButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_RESTART_GAME]];
	[mainMenuButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_MAINMENU]];
	[helpButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAME_TUTORIAL]];
}

-(void) onHelpPressed {
    if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	CCScene* scene = [CCBReader sceneWithNodeGraphFromFile:@"HelpScene.ccbi"];
	[[CCDirector sharedDirector] pushScene:scene];
}

-(void) onRestartGamePressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
    [[CCDirector sharedDirector].touchDispatcher removeDelegate:self];
	[self removeFromParentAndCleanup:YES];
	if (self.delegate) [self.delegate restartGame];
}

-(void) onMainMenuPressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
    [[CCDirector sharedDirector].touchDispatcher removeDelegate:self];
	[self removeFromParentAndCleanup:YES];
	if (self.delegate) [self.delegate toMainMenu];
}

@end
